modules/govuk/manifests/apps/govuk_content_schemas.pp

```diff
@@ -4,17 +4,31 @@
 #
 # === Parameters
 #
+# [*ensure*]
+#   Allow schemas to be removed.
+#
 # [*directory*]
 #   The directory in which the govuk_content_schemas should be
-#   available.
+#   deployed.
 #
 class govuk::apps::govuk_content_schemas(
-  $directory = '/data/apps/govuk-content-schemas/current',
+  $ensure = present,
+  $directory = '/data/apps/govuk-content-schemas',
 ) {
   include icinga::client::check_directory_exists
 
+  if $ensure == absent {
+    file { $directory:
+      ensure => absent,
+      force  => true,
+    }
+  }
+
+  $current_directory = "${directory}/current"
+
   @@icinga::check { "check_govuk_content_schemas_on_${::hostname}":
```

Comment:
I think this check is automatically managed, so removing the `ensure` and putting it inside an `if $ensure == present { ... }` should do the job.

Reply:
Sure, do you think there are any advantages to doing it that way rather than the approach I've took? I have a preference for passing in the ensure since it's more succinct and not nested but not sure if I'm overlooking something.

Reply:
I thought this hadn't worked, because the alert was still there after 25 minutes, but it's now gone away.  I guess puppet had to run somewhere else too (the monitoring box?) to clean up the old alerts.

Reply:
Ah good good, thank you for trying to help solve the problem - even if it turned out not to be one. I appreciate the effort 🙂.

Yeah I've realised I'm not really sure how these icinga checks work - didn't realise we could trigger an object in one node that affects another. Everyday is learning day.

Reply:
Yeah, while Puppet should have run once on every machine within 30 minutes, some changes like Icinga checks can require Puppet to run on say a backend machine to change something about a check, and then on the monitoring machine to actually change the Icinga configuration.